packages/core/src/odata/common/selectable/one-to-many-link.ts

```diff
@@ -30,9 +30,18 @@ export class OneToManyLink<
   /**
    * @experimental This is experimental and is subject to change. Use with caution.
    */
-  filter(...expressions: Filterable<LinkedEntityT>[]): this {
+  filter(
+    ...expressions: (
+      | Filterable<LinkedEntityT>
+      | OneToManyLink<LinkedEntityT, any>
+    )[]
+  ): this {
     const link = this.clone();
-    link._filters = new FilterLink(this, expressions);
+
+    link._filters = new FilterLink(
+      this,
+      expressions.map(e => (e instanceof OneToManyLink ? e._filters : e))
```

Comment:
just to be sure. Will this work sort of recursively if you would add a third layer:
```
  TestEntity.TO_MULTI_LINK_1.filter(
            TestEntityMultiLink_1.TO_MULTI_LINK_2.filter(
                 TestEntityMultiLink_2.TO_MULTI_LINK_3.filter(
...
```
with an if you could get two levels but the third not. Perhaps it is already working. Just want to be sure.

Reply:
Please keep in mind, the same function  names `filter()` are used in difference classes, which makes it confusing. There is a ticket:
https://github.wdf.sap.corp/MA/scp-rocket/issues/1287

```
People.requestBuilder()
      .getAll()
      .expand(
        People.FRIENDS.filter( [1]
          People.FRIENDS.filter( [2]
            any(People.USER_NAME.equals('russellwhyte'))
          )
        )
      )
```
will generate url like
```
$expand=Friends($filter=Friends/any(a0:a0/UserName eq 'russellwhyte'))
```
so, the [1] filter is actually a "setter" for setting the filter of the expanded navigation property.
the [2] filter is static method for building the lambda expression.

Therefore, there are only two levels.
Using the same function for different purpose is not good. Since this is the agreed api design, we consider as the initial version. The follow up ticket should solve it.

Reply:
ok